dll/ntdll/csr/connect.c

```diff
@@ -62,10 +62,15 @@ CsrpConnectToServer(IN PWSTR ObjectDirectory)
     /* Calculate the total port name size */
     PortNameLength = ((wcslen(ObjectDirectory) + 1) * sizeof(WCHAR)) +
                      sizeof(CSR_PORT_NAME);
+    if (PortNameLength > UNICODE_STRING_MAX_CHARS)
+    {
+        DPRINT1("PortNameLength too big: %Iu", PortNameLength);
+        return STATUS_INTEGER_OVERFLOW;
```

Comment:
The standard status code in this precise situation, is to return `STATUS_NAME_TOO_LONG` instead.

Reply:
fixed